android/src/com/mapswithme/maps/settings/StoragePathManager.java

```diff
@@ -228,13 +223,6 @@ private void updateExternalStorages(Application application)
           continue;
         }
 
-        final long freeSize = StorageUtils.getFreeBytesAtPath(path);
-        if (freeSize <= 0)
```

Comment:
Why read-only partitions are needed?

Reply:
This is for a case of a full storage when the download had been interrupted because of not enough space - the "save maps to" menu disappeared for me, which is confusing.
A user should see options with no space to be aware of them so that the space could be freed.

I'm not sure yet if such storages will pass the next "writability" test though.

Reply:
Changing storage (WritableDir) affects not only maps downloading, but also saving bookmarks. Or not? Should we allow it without ability to save bookmarks later?

Reply:
At the moment bookmarks are saved to internal memory always.
Actually I didn't test yet if e.g. disappear of SD card changes saved "StoragePath" in settings file.
(though in the code I touched its not saved, but just an error is logged if configured path is not in the storage options)

Reply:
Yes, you are right:
```
std::string GetBookmarksDirectory()
{
  return base::JoinPath(GetPlatform().SettingsDir(), "bookmarks");
}

```

Reply:
> 
> I'm not sure yet if such storages will pass the next "writability" test though.

Indeed, an SD card with 0 free space doesn't pass an additional mkdir() test.
So I'm going to remove this extra writeability check because it seems it was made for 4.4 devices specifically. I searched for similar canWrite() issues for later android versions and found nothing.
Anyways our code that does the actual writing should catch exceptions if any (I've tested its working for map downloads/updates).

Reply:
The logic should be different. Our code should always show the storage *which is in our settings file* despite its free space.

Reply:
Yeap, this is how its going to be after this PR.